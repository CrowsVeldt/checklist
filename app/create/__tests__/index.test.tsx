import { render } from "@testing-library/react-native";
import CreateChecklist from "../index";
import AppProvider from "@/context/AppContext";

describe("<CreateChecklist />", () => {
  test("Index renders correctly", () => {
    const tree = render(
      <AppProvider>
        <CreateChecklist />
      </AppProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
