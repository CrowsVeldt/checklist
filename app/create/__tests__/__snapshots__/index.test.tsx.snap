// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateChecklist /> Index renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        {
          "padding": 10,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": "20%",
        }
      }
    >
      Title:
    </Text>
    <TextInput
      maxLength={16}
      onChangeText={[Function]}
      style={
        {
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 1,
          "width": "60%",
        }
      }
      value="Title"
    />
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 10,
            "width": "70%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          value="Item 1"
        />
        <View
          style={
            {
              "alignItems": "center",
            }
          }
        >
          <Text>
            Required?
          </Text>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "borderColor": "#657786",
                  "borderRadius": 2,
                  "borderWidth": 2,
                  "height": 20,
                  "width": 20,
                },
                {
                  "borderRadius": 5,
                },
                false,
                false,
                undefined,
                false,
              ]
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={{}}
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderColor": "black",
          "borderWidth": 1,
        }
      }
    >
      <Text>
        +
      </Text>
    </View>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderColor": "black",
          "borderRadius": 5,
          "borderWidth": 1,
          "padding": 20,
        }
      }
    >
      <Text>
        Create
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
